debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/VmCheckCommand.java

```diff
@@ -236,13 +236,25 @@ private void checkJ9ClassSanity(J9JavaVMPointer javaVM, PrintStream out) throws
 
 	private boolean verifyJ9Class(J9JavaVMPointer javaVM, PrintStream out, J9ClassPointer classPointer) throws CorruptDataException {
 		boolean passed = verifyJ9ClassHeader(javaVM, out, classPointer);
+		J9ClassLoaderPointer classLoader;
 
-		if (!classPointer.classLoader().isNull()) {
-			J9MemorySegmentPointer segment = classPointer.classLoader().classSegments();
+		if (J9ROMClassHelper.isAnonymousClass(classPointer.romClass())) {
+			try {
+				classLoader = javaVM.anonClassLoader();
+			} catch (NoSuchFieldException e) {
+				// Should not happen
+				classLoader = classPointer.classLoader();
+			}
```

Comment:
@keithc-ca Is this what you were looking for?

Reply:
If that's not expected to happen, perhaps a better response is to throw `CorruptDataException`. The test above concluded the class is anonymous, so it should be from a VM that has the `anonClassLoader` field.
Alternatively, we can examine git history: if we can confirm that the field has been present in every revision since openj9 was open-sourced, then the field can be marked as 'required' and the exception will not be thrown.

Reply:
According to https://github.com/eclipse-openj9/openj9/blame/c306d8cebd03961fab54adf851cae2cf5ac8fabe/runtime/oti/j9nonbuilder.h#L5463, the declaration of `anonClassLoader` was present in the initial open-sourced revision and has not been modified since. In which case, the field should be marked as 'required'; the generated method will not throw `NoSuchFieldException` and so need not be caught here.